src/tableView.js

```diff
@@ -263,15 +263,16 @@ function TableView(instance) {
     },
     onCellContextMenu: (event, coords, TD, wt) => {
       that.activeWt = wt;
-      instance.runHooks('beforeOnCellContextMenu', event, coords, TD);
 
       if (instance.selection.isInProgress()) {
         instance.selection.finish();
-
-        isMouseDown = false;
       }
 
+      isMouseDown = false;
+
+      instance.runHooks('beforeOnCellContextMenu', event, coords, TD);
```

Comment:
To be more compatible with similar hooks could you move `beforeOnCellContextMenu` hook at the beginning of the `onCellContextMenu` function? As you had before. Moreover, it could be good an idea to add support for stoping event propagation: 
```
if (isImmediatePropagationStopped(event)) {
    return;
}
```
This solution will force you to move context menu logic to `afterOnCellContextMenu` hook. 

Looking through similar functionalities (sorting, filtering) all of them are fired right before `after*` hooks but never directly in `before*` hooks this gives an opportunity to cancel that features directly in the `before*` hooks. The same should be achieved here.

Reply:
👍